apis/Google.Cloud.Spanner.V1/Google.Cloud.Spanner.V1/SessionPoolImpl.cs

```diff
@@ -130,7 +143,7 @@ private bool TryPop(TransactionOptions options, out SessionPoolEntry entry)
                             {
                                 Logger.Debug(() => "found a session with matching transaction semantics.");
                                 indexToUse = i;
-                                if (options.ModeCase == TransactionOptions.ModeOneofCase.ReadOnly
+                                if (options?.ModeCase != TransactionOptions.ModeOneofCase.ReadWrite
```

Comment:
Possibly rename the local variable to `transactionOptions` to avoid having both `_options` and `options` in scope? (Or rename `_options` to `_poolOptions`?)